core/src/integrationTest/kotlin/ch/ergon/dope/clauses/JoinIntegrationTest.kt

```diff
@@ -0,0 +1,90 @@
+import ch.ergon.dope.integrationTest.TestCouchbaseDatabase.testBucket
+import ch.ergon.dope.integrationTest.toMapValues
+import ch.ergon.dope.integrationTest.tryUntil
+import ch.ergon.dope.resolvable.clause.model.joinHint.HashOrNestedLoopHint.NESTED_LOOP
+import ch.ergon.dope.resolvable.clause.model.joinHint.indexHint
+import ch.ergon.dope.resolvable.expression.alias
+import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.resolvable.expression.unaliased.type.meta.meta
+import ch.ergon.dope.resolvable.expression.unaliased.type.relational.isEqualTo
+import ch.ergon.dope.validtype.StringType
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JoinIntegrationTest : BaseIntegrationTest() {
+    @Test
+    fun `join all orders on employee id`() {
+        val employeeAlias = testBucket.alias("e")
+        val orderAlias = testBucket.alias("o")
+        val employeeNameField = Field<StringType>("name", employeeAlias.alias)
+        val orderNumberField = Field<StringType>("orderNumber", orderAlias.alias)
+        val orderEmployeeIdField = Field<StringType>("employee", orderAlias.alias)
+        val dopeQuery = QueryBuilder()
+            .select(
+                employeeNameField,
+                orderNumberField,
+            )
+            .from(
+                employeeAlias,
+            )
+            .join(
+                orderAlias,
+                orderEmployeeIdField.isEqualTo(meta(employeeAlias).id),
+            )
+            .build()
+
+        tryUntil {
+            val queryResult = queryWithoutParameters(dopeQuery)
+            val result = queryResult.toMapValues()
+
+            assertEquals(5, queryResult.rows.size)
+            assertEquals("employee1", result["name"])
+            assertEquals("order1", result["orderNumber"])
+        }
+    }
+
+    @Test
+    fun `join documents with multiple joins`() {
+        val employeeAlias = testBucket.alias("e")
+        val clientAlias = testBucket.alias("c")
+        val orderAlias = testBucket.alias("o")
+        val employeeNameField = Field<StringType>("name", employeeAlias.alias)
+        val clientNameField = Field<StringType>("name", clientAlias.alias)
+        val orderNumberField = Field<StringType>("orderNumber", orderAlias.alias)
+        val orderEmployeeIdField = Field<StringType>("employee", orderAlias.alias)
+        val orderClientIdField = Field<StringType>("client", orderAlias.alias)
+        val dopeQuery = QueryBuilder()
+            .select(
+                orderNumberField,
+                employeeNameField.alias("employeeName"),
+                clientNameField.alias("clientName"),
+            )
+            .from(
+                orderAlias,
+            )
+            .join(
+                employeeAlias,
+                orderEmployeeIdField.isEqualTo(meta(employeeAlias).id),
+                hashOrNestedLoopHint = NESTED_LOOP,
+            ).innerJoin(
```

Comment:
add line break for consistency. 
```
)
.innerJoin
```

Or what i would prefer that you do it like it already is but chane it everywhere else.
So that it follows this pattern:
```
QueryBuilder()
    .select(
        ...
    ).from(...) // when the closing bracket is on a newline add the function call on the same line
    .join // when the closing and opening brackets are on the same line add the function call on a new line
...
```

Reply:
check all tests